Objects/longobject.c

```diff
@@ -3038,17 +3038,14 @@ long_compare(PyLongObject *a, PyLongObject *b)
     }
     else {
         Py_ssize_t i = Py_ABS(Py_SIZE(a));
-        while (--i >= 0 && a->ob_digit[i] == b->ob_digit[i])
+        while (--i > 0 && a->ob_digit[i] == b->ob_digit[i])
             ;
-        if (i < 0)
-            sign = 0;
-        else {
-            sign = (sdigit)a->ob_digit[i] - (sdigit)b->ob_digit[i];
-            if (Py_SIZE(a) < 0)
-                sign = -sign;
+        sign = (sdigit)a->ob_digit[i] - (sdigit)b->ob_digit[i];
+        if (Py_SIZE(a) < 0) {
+            sign = -sign;
         }
     }
-    return sign < 0 ? -1 : sign > 0 ? 1 : 0;
```

Comment:
long_compare() is only accessible in longobject.c. I confirm that this change is safe: result == -1 or result == 1 is never used in longobject.c, only comparisons like result == 0, result < 0, result >= 0, etc.

Reply:
Oh, "return sign" implicitly downcasts Py_ssize_t to int. I suggst to change long_compare() result type from int to Py_ssize_t, which requires to modify some calls to long_compare() (change the type of the variable used to store the result).

Reply:
This has been resolved on new change.